scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,238 @@
+        result = result.convert()
+        return int(result['results']['bindings'][0]['total']['value'])
+
+    def get_uris(self, start_date, end_date, limit, offset):
+        uris = []
+        query_str = """ PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                        PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+
+                        SELECT ?uri
+                        {{
+                            ?uri vivo:dateTimeValue ?dateUri .
+                            ?dateUri vivo:dateTime ?date .
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date)
+                        }}  LIMIT {} OFFSET {}
+                    """
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        for result in results['results']['bindings']:
+            uris.append(result['uri']['value'])
+        return uris
+
+    def complete_authors(self, authors):
+        for author in authors:
+            pattern = """ <{}> obo:ARG_2000028 ?vcard .
+                          ?vcard vcard:hasEmail ?emailUri .
+                          ?emailUri vcard:email ?email ."""
+            email = self.get_string(author['sameAs'], {'name': 'email', 'pattern': pattern, 'type': 'string'})
+            if email:
+                author['email'] = email
+
+            pattern = """?role obo:RO_0000052 <{}> .
+                         ?role vivo:roleContributesTo ?organization .
+                         ?organization rdfs:label ?name .
+                         ?role vivo:dateTimeInterval ?interval .
+                        FILTER NOT EXISTS {{ ?interval vivo:end ?end }}"""
+            affiliation = self.get_dict(author['sameAs'], {'name': 'affiliation', 'fields': ['name'], 'pattern': pattern, 'type': 'dict'})
+            if affiliation:
+                author['affiliation'] = affiliation
+
+            pattern = """<{}> vivo:orcidId ?orcidId .
+                        FILTER isURI(?orcidId)"""
+            orcidId = self.get_string(author['sameAs'], {'name': 'orcidId', 'pattern': pattern, 'type': 'string'})
+            author['sameAs'] = [author['sameAs']]
+            if orcidId:
+                author['sameAs'].append(orcidId)
+        return authors
+
+    @property
+    def schema(self):
+        return {
+            'title': ('/title', lambda x: x if x else ''),
+            'providerUpdatedDateTime': ('/date', lambda x: datetime.strptime(x, "%Y-%m-%d").strftime("%Y-%m-%dT%H:%M:%S%Z") + '+00:00'),
+            'uris': {
+                'canonicalUri': '/uri',
+                'providerUris': ['/uri'],
+                'objectUris': ('/pmid', '/doi', lambda x, y: process_object_uris(x, y))
+            },
+            'contributors': ('/authors', lambda x: self.complete_authors(x)),
```

Comment:
see above, passing in `lambda x: my_function(x)` is equivalent to just passing in `my_function`